recipes/ghcfilesystem/all/conanfile.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+
+class GhcFilesystemRecipe(ConanFile):
+    name = "ghcfilesystem"
+    description = "A header-only single-file std::filesystem compatible helper library"
+    topics = ("conan", "bhcfilesystem", "header-only", "filesystem")
+    homepage = "https://github.com/gulrak/filesystem"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    no_copy_source = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "filesystem-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["GHC_FILESYSTEM_BUILD_TESTING"] = False
+        self._cmake.definitions["GHC_FILESYSTEM_BUILD_EXAMPLES"] = False
+        self._cmake.definitions["GHC_FILESYSTEM_WITH_INSTALL"] = True
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib"))
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "ghcFilesystem"
+        self.cpp_info.names["cmake_find_package_multi"] = "ghcFilesystem"
+        self.cpp_info.components["filesystem"].names["cmake_find_package"] = "ghc_filesystem"
+        self.cpp_info.components["filesystem"].names["cmake_find_package_multi"] = "ghc_filesystem"
```

Comment:
If you take a look on its CMakeLists.txt. The project exports `ghcFilesystem::ghc_filesystem`. The component it's a trick to inject that target name, otherwise we would have `ghcFilesystem::ghcFilesystem` only. We try to reproduce as closer as possible to the upstream.

Reply:
Ah ok, I obviously overlooked this part. I will check, hope then I understand the original motivation. I definitley agree with "We try to reproduce as closer as possible to the upstream"!

Reply:
Here: https://github.com/gulrak/filesystem/blob/v1.4.0/CMakeLists.txt#L58

Reply:
I see, this line and the `add_library(ghc_filesystem INTERFACE)` in combination I guess.

Reply:
If you where to clone and run `cmake .. && cmake --install .`

You can consume the project with

```cmake
find_package(ghcFilesystem)
...
target_link_library(example PRIVATE ghcFilesystem::ghc_filesystem)
```

All because of the few line that uilianries pointed out

You can add it to the test package with the `cmake_find_package` generator to test this out

Reply:
I need to refresh of my CMake knowledge... Thanks for explaining!